x-pack/plugins/security_solution/public/timelines/components/timeline/body/helpers.tsx

```diff
@@ -109,29 +107,38 @@ export const getEventType = (event: Ecs): Omit<EventType, 'all'> => {
   return 'raw';
 };
 
-export const isInvestigateInResolverActionEnabled = (ecsData?: Ecs) => {
-  return (
-    get(['agent', 'type', 0], ecsData) === 'endpoint' &&
-    get(['process', 'entity_id'], ecsData)?.length === 1 &&
-    get(['process', 'entity_id', 0], ecsData) !== ''
-  );
-};
+export const isInvestigateInResolverActionEnabled = (ecsData?: Ecs) =>
+  get(['agent', 'type', 0], ecsData) === 'endpoint' &&
+  get(['process', 'entity_id'], ecsData)?.length === 1 &&
+  get(['process', 'entity_id', 0], ecsData) !== '';
 
-export const getInvestigateInResolverAction = ({
-  dispatch,
-  timelineId,
-}: {
-  dispatch: Dispatch;
+interface InvestigateInResolverActionProps {
   timelineId: string;
-}): TimelineRowAction => ({
-  ariaLabel: i18n.ACTION_INVESTIGATE_IN_RESOLVER,
-  content: i18n.ACTION_INVESTIGATE_IN_RESOLVER,
-  dataTestSubj: 'investigate-in-resolver',
-  displayType: 'icon',
-  iconType: 'node',
-  id: 'investigateInResolver',
-  isActionDisabled: (ecsData?: Ecs) => !isInvestigateInResolverActionEnabled(ecsData),
-  onClick: ({ eventId }: TimelineRowActionOnClick) =>
-    dispatch(updateTimelineGraphEventId({ id: timelineId, graphEventId: eventId })),
-  width: DEFAULT_ICON_BUTTON_WIDTH,
-});
+  ecsData: Ecs;
+}
+
+export const InvestigateInResolverAction = React.memo<InvestigateInResolverActionProps>(
+  ({ timelineId, ecsData }) => {
+    const dispatch = useDispatch();
+    const isDisabled = useMemo(() => !isInvestigateInResolverActionEnabled(ecsData), [ecsData]);
+    const handleClick = useCallback(
+      ({ eventId }) =>
+        dispatch(updateTimelineGraphEventId({ id: timelineId, graphEventId: eventId })),
```

Comment:
Seeing a bug where the _Analyze Event_ view doesn't appear when the icon is clicked. I set a breakpoint here, and it looks like the issue is `eventId` is `undefined`.

To reproduce:

1) create a new timeline
2) enter the following KQL: `agent.type: endpoint and process.entity_id: *`
3) optionally expand the time range (if necessary)
4) click the `Analyze event` action

**Expected result**
The Analyze event view is displayed

Reply:
Great catch @andrew-goldstein 🕵️